samples/dect/dect_phy/dect_shell/src/dect/mac/dect_phy_mac_client.c

```diff
@@ -196,7 +202,73 @@ static uint64_t dect_phy_mac_client_next_rach_tx_time_get(
 	return ra_start_mdm_ticks;
 }
 
-int dect_phy_mac_client_rach_tx(struct dect_phy_mac_nbr_info_list_item *target_nbr,
+static int dect_phy_mac_client_rach_tx(struct dect_phy_mac_nbr_info_list_item *target_nbr,
+				struct dect_phy_mac_rach_tx_params *params);
+
+struct dect_phy_mac_client_rach_tx_data {
+	struct k_work_delayable work;
+	struct dect_phy_mac_rach_tx_params cmd_params;
+	struct dect_phy_mac_nbr_info_list_item target_nbr;
+};
+static struct dect_phy_mac_client_rach_tx_data client_rach_tx_work_data;
+
+static void dect_phy_mac_client_rach_tx_worker(struct k_work *work_item)
+{
+	struct k_work_delayable *delayable_work = k_work_delayable_from_work(work_item);
+	struct dect_phy_mac_client_rach_tx_data *data =
+		CONTAINER_OF(delayable_work, struct dect_phy_mac_client_rach_tx_data, work);
+	struct dect_phy_mac_rach_tx_params cmd_params;
+	int err;
+
+	cmd_params = data->cmd_params;
+
+	if (data->cmd_params.get_mdm_temp) {
+		char tmp_str[DECT_DATA_MAX_LEN * 2] = {0};
+		int mdm_temperature = dect_phy_ctrl_modem_temperature_get();
+
+		/* Modem temperature to be included in a data JSON */
+		if (mdm_temperature == NRF_MODEM_DECT_PHY_TEMP_NOT_MEASURED) {
+			sprintf(tmp_str, "{\"data\":\"%s\",\"m_tmp\":\"N/A\"",
+				cmd_params.tx_data_str);
+		} else {
+			sprintf(tmp_str, "{\"data\":\"%s\",\"m_tmp\":\"%d\"",
+				cmd_params.tx_data_str, mdm_temperature);
+		}
+		strcat(tmp_str, "}");
```

Comment:
Is there are reason why this is in a separate `strcat()` instead of the `sprintf()` calls above?

Reply:
Yes, it could be there as well. I'll change it.